rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/names.hpp

```diff
@@ -15,27 +15,27 @@
 #ifndef RMW_FASTRTPS_SHARED_CPP__NAMES_HPP_
 #define RMW_FASTRTPS_SHARED_CPP__NAMES_HPP_
 
+#include <string>
+
 #include "fastrtps/utils/fixed_size_string.hpp"
 #include "rmw/types.h"
 #include "namespace_prefix.hpp"
 
-/// Construct a topic name according to proper conventions.
+/// Construct a topic name.
 /**
-  * \param[in] qos_profile The QoS profile for the topic.
   * \param[in] prefix Required prefix for topic name.
   * \param[in] base Required name of the topic.
   * \param[in] suffix Optional suffix for topic name.
   */
 inline
-eprosima::fastrtps::string_255
-_create_topic_name(
-  const rmw_qos_profile_t * qos_profile,
+std::string
```

Comment:
Why change away from the string_255? I think that was done to avoid heap allocations and for performance.

Reply:
I needed `_mangle_topic_name` to return an string, as some `GraphCache` methods expect that.

----

Bottom line:
I actually regret not having had adding a template parameter for that in [GraphCache](https://github.com/ivanpauno/rmw_dds_common/blob/c56f0492191f0f4eb8d08ce9cdeb7cbabf4ba106/rmw_dds_common/include/rmw_dds_common/graph_cache.hpp#L48).
Probably, I could have also added a template parameter for the GUID, instead of transforming everything in a `rmw_gid_t`.
Moreover, it would make sense to make `GraphCache` implementation header only, so it can be optimized when compiled with an specific rmw implementation.

Having say that, I think that it makes sense to not worry about that right now, and iterate on those details right after merging this (and before refactoring another rmw implementation).

Reply:
I don't follow exactly what doesn't work in this case, but maybe it's worth following up on.

Can you not do `std::string s = <some func that returns string_255>`? i.e. is there no assignment conversion for `std::string`?

Reply:
Applied that in https://github.com/ros2/rmw_fastrtps/pull/312/commits/332c7bb7956522c7cb78573b7b2451b7354cfcc5.

----

I can't do the same change in the demangle functions https://github.com/ros2/rmw_fastrtps/blob/bd5507b61c7285d6083e514ccd49f00b2945baeb/rmw_fastrtps_shared_cpp/src/demangle.hpp#L21 (they were already returning a `std::string`).

The reason is that `GraphCache` expects a demangle function to be called with the following signature: https://github.com/ivanpauno/rmw_dds_common/blob/c56f0492191f0f4eb8d08ce9cdeb7cbabf4ba106/rmw_dds_common/include/rmw_dds_common/graph_cache.hpp#L326.
I used `std::string` because the idea is to reuse `GraphCache` in others rmw implementations, but of course, it can be solved by templetazing the class.

Reply:
Ok sounds good.